web/src/test/java/org/openmrs/web/filter/update/GZIPFilterTest.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.filter.update;
+
+import com.sun.xml.ws.server.UnsupportedMediaException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mockito;
+import org.openmrs.GlobalProperty;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.Verifies;
+import org.openmrs.web.filter.GZIPFilter;
+import org.openmrs.web.test.BaseWebContextSensitiveTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.util.zip.GZIPOutputStream;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests some of the methods on the {@link org.openmrs.web.filter.update.UpdateFilter}
+ */
+public class GZIPFilterTest extends BaseWebContextSensitiveTest {
+	
+	/**
+	 * @see {@link org.openmrs.web.filter.GZIPFilter#performGZIPRequest(HttpServletRequest,HttpServletResponse, javax.servlet.FilterChain)}
+	 */
+	@Test
+	@Verifies(value = "zip request and response", method = "performGZIPRequest(HttpServletRequest,HttpServletResponse,FilterChain)")
+	public void ZipRequestTest() throws Exception {
```

Comment:
This test should be renamed to something more meaningful. Also methods in Java start with lowercase.